### ISSUE TITLE
Project: SharePoint Page toolkit

### ISSUE THREAD

**Issue by tarzzi** [microsoft/hack-together#118]
### Project name

SPPage toolkit

## About 
SPPage toolkit is an simple toolset for basic CRUD operations for SharePoint site pages.

## Type of app?
Solution is built on .NET MAUI, since the subject was new to me (only a tiny bit of WPF knowledge prior) and cross platform possiblities are always exiting.

Built on Graph Beta endpoint, since Pages is [still in beta](https://learn.microsoft.com/en-us/graph/api/resources/sitepage?view=graph-rest-beta) as of 3/2023.

## What does it do?
It allows the user to:
1. Search for a SharePoint site with given term
2. List all the pages that are inside that site
3. Gives possibilities to create, update and delete pages as needed

## What problem does it solve?
You know the feeling when you're solving a client issue and for each demo, test or repro, you create a new site page? And they keep cluttering up your sitepages list?

So do I. 

But quite rarely, since my dev tenant is a graveyard of sitepages.

So this toolkit was created to easily manage your site pages, and click away pages that are no longer needed, or to create a bunch of new ones or to even rename that silly old Dog named page to a Cat page! 

### Repo URL

https://github.com/tarzzi/msoft-hack-together-2023

### Team members

Tarzzi

**Comment by waldekmastykarz** [comment]
Thank you for your submission @tarzzi! Would you be willing to record a short video to demonstrate the app? It's not required for the hackathon but it would help us showcase your work internally at Microsoft and publicly in the community 😊

**Comment by tarzzi** [comment]
Of course, will do! 😊 Is youtube fine or are there any preferences? @waldekmastykarz

**Comment by waldekmastykarz** [comment]
YouTube, here or in the repo, anything will work. Thank you!

**Comment by tarzzi** [comment]
[Demo](https://github.com/tarzzi/msoft-hack-together-2023/blob/d78841096e872bc1efe7aebe7c726c9ac38d68e0/sptools_demo.mp4) available at repo! 📺
